onnx/defs/training/defs.cc

```diff
@@ -196,4 +196,150 @@ ONNX_TRAINING_OPERATOR_SET_SCHEMA(
              "tensor(double)"},
             "Allow inputs to be any kind of floating-point tensor."));
 
+
+static const char* GraphCall_ver1_doc = R"DOC(
+The GraphCall operator invokes a graph inside TrainingInfoProto's
+algorithm field. The mapping from GraphCall inputs to the invoked graph's
+input list is determined by the attributes "input_names" and "output_names".
+Assume that ModelProto's graph field has
+
+- name: MyInferenceGraph
+- inputs: [X, W, Z]
+- initializer: [W]
+- outputs: Y
+
+as visualized bellow.
+
+```
+X -----.
+       |
+       v
+W --> Conv --> H --> Gemm --> Y
+                      ^
+                      |
+                      Z
+```
+
+Assume that the training algorithm contains
+
+- inputs: [X_1, Z_1, L]
+- outputs: [W_new]
+
+Inside the training algorithm graph, one can invoke the graph via adding
+a GraphCall node with
+
+- inputs: [X_1, Z_1]
+- outputs: [Y_1]
+- an attribute graph_name="MyInferenceGraph",
+- an attribute input_names=["X", "Z"]. It means the first input "X_1"
+  should be bound to "X" in the inference graph. Similarly, "Z_1" may
+  be bound to "Z" in the inference graph. Note that the value of "W" is
+  not needed because "W" is also an initializer in the inference graph.
+- an attribute output_names=["Y"] which means the inference graph's
+  output "Y" should be used as "Y_1" in the training algorithm.
```

Comment:
The spec says this for nodes:
> Input and outputs are positionally associated with operator inputs and outputs. Attributes are associated with operator attributes by name.

I think it would make sense to follow the same convention for `GraphCall` of positionally associating inputs and outputs. Optional inputs can be left out by providing `""`, and the same works for outputs.
E.g.
```python
onnx.helper.make_node(
    op_type= "GraphCall",
    inputs = ["X_1", "", "Z_1"],
    outputs = ["Y_1"], # Could use ["", "Y_1"] if there were two outputs
    # Attributes:
    graph_name = "MyInferenceGraph",
)
```

Reply:
Ok. Let's follow this align-by-position convention. Thanks for pointing this out.